sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -222,7 +222,7 @@ case class AdaptiveSparkPlanExec(
       .map(_.toLong).filter(SQLExecution.getQueryExecution(_) eq context.qe)
   }
 
-  private def getFinalPhysicalPlan(): SparkPlan = lock.synchronized {
+  private[sql] def getFinalPhysicalPlan(): SparkPlan = lock.synchronized {
```

Comment:
looking at the code again, I think it's risky to call `getFinalPhysicalPlan` directly. Other places call `withFinalPlanUpdate` which can handle UI update. How about we add a new function here?
```
def finalPhysicalPlan = withFinalPlanUpdate(identity)
```